client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcPreparedStatement.java

```diff
@@ -1051,128 +1074,225 @@ public void testConvertTimestamp()
+        assertBind((ps, i) -> ps.setObject(i, "1970-01-01 12:34:56.123456789012 +05:45", Types.TIMESTAMP_WITH_TIMEZONE), useLegacyPreparedStatements)
                 .resultsIn("timestamp(12) with time zone", "TIMESTAMP '1970-01-01 12:34:56.123456789012 +05:45'");
     }
 
-    @Test
-    public void testInvalidConversions()
+    @Test(dataProvider = "legacyPreparedStatementProvider")
+    public void testInvalidConversions(boolean useLegacyPreparedStatements)
             throws SQLException
     {
-        assertBind((ps, i) -> ps.setObject(i, String.class)).isInvalid("Unsupported object type: java.lang.Class");
-        assertBind((ps, i) -> ps.setObject(i, String.class, Types.BIGINT)).isInvalid("Cannot convert instance of java.lang.Class to SQL type " + Types.BIGINT);
-        assertBind((ps, i) -> ps.setObject(i, "abc", Types.SMALLINT)).isInvalid("Cannot convert instance of java.lang.String to SQL type " + Types.SMALLINT);
+        assertBind((ps, i) -> ps.setObject(i, String.class), useLegacyPreparedStatements).isInvalid("Unsupported object type: java.lang.Class");
+        assertBind((ps, i) -> ps.setObject(i, String.class, Types.BIGINT), useLegacyPreparedStatements)
+                .isInvalid("Cannot convert instance of java.lang.Class to SQL type " + Types.BIGINT);
+        assertBind((ps, i) -> ps.setObject(i, "abc", Types.SMALLINT), useLegacyPreparedStatements)
+                .isInvalid("Cannot convert instance of java.lang.String to SQL type " + Types.SMALLINT);
+    }
+
+    @Test
+    public void testLegacyPreparedStatementsTrue()
+            throws Exception
+    {
+        testLegacyPreparedStatementsSetting(true,
+                "EXECUTE statement1 USING INTEGER '42', 'value1''s'");
     }
 
-    private BindAssertion assertBind(Binder binder)
+    @Test
+    public void testLegacyPreparedStatementsFalse()
+            throws Exception
     {
-        return new BindAssertion(this::createConnection, binder);
+        testLegacyPreparedStatementsSetting(false,
+                "EXECUTE IMMEDIATE 'SELECT * FROM blackhole.blackhole.test_table WHERE x = ? AND y = ? AND y <> ''Test''' USING INTEGER '42', 'value1''s'");
     }
 
-    private Connection createConnection()
+    private BindAssertion assertBind(Binder binder, boolean useLegacyPreparedStatements)
+    {
+        return new BindAssertion(() -> this.createConnection(useLegacyPreparedStatements), binder);
+    }
+
+    private Connection createConnection(boolean useLegacyPreparedStatements)
             throws SQLException
     {
         String url = format("jdbc:trino://%s", server.getAddress());
-        return DriverManager.getConnection(url, "test", null);
+        Properties properties = new Properties();
+        properties.setProperty("user", "test");
+        properties.setProperty("legacyPreparedStatements", String.valueOf(useLegacyPreparedStatements));
+        return DriverManager.getConnection(url, properties);
```

Comment:
You can simply add the new property to the URL
```
String url = format("jdbc:trino://%s?legacyPreparedStatements=" + legacyPreparedStatements, server.getAddress());
return DriverManager.getConnection(url, "test", null);
```

Reply:
Done